scene/gui/text_edit.cpp

```diff
@@ -1415,6 +1420,24 @@ void TextEdit::_notification(int p_what) {
 							}
 						}
 
+						// Draw an underline to give more accurate information about the mark's location.
+						// The background line is drawn on a per-line basis, so it's done above (outside this loop).
+						if (text.get_mark_position(line) >= 2) {
+
+							// The mark provides column information.
+							if (j >= text.get_mark_position(line) - 1) {
+								VisualServer::get_singleton()->canvas_item_add_rect(ci, Rect2(Point2i(char_ofs + char_margin + ofs_x, ofs_y + get_row_height() - 2), Size2i(char_w, 2)), Color(cache.mark_color.r, cache.mark_color.g, cache.mark_color.b, 0.8));
+							}
+						} else if (text.get_mark_position(line) == 1) {
+
+							// The mark doesn't provide column information (it's on the first character).
+							// We consider tabs as spaces for the drawing offset.
+							// Otherwise, the line would start too late when using tab indentation.
+							if (j >= get_indent_level(line, true)) {
+								VisualServer::get_singleton()->canvas_item_add_rect(ci, Rect2(Point2i(char_ofs + char_margin + ofs_x, ofs_y + get_row_height() - 2), Size2i(char_w, 2)), Color(cache.mark_color.r, cache.mark_color.g, cache.mark_color.b, 0.8));
+							}
+						}
+
```

Comment:
Can call `_find_first_non_whitespace_column_of_line` instead of changing `get_indent_level` for the column number.

In addition to this, not a fan of calculating this this on every char. Can pre-calculate the column above if the line is marked, similar to `search_text_col` / `highlighted_text_col` etc. Then we only need to check the column here.